user/user_validator.go

```diff
@@ -0,0 +1,147 @@
+// Copyright © 2020 Intel Corporation
+//
+// SPDX-License-Identifier: GPL-3.0-only
+
+package user
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/clearlinux/clr-installer/cmd"
+	"github.com/clearlinux/clr-installer/log"
+)
+
+// Validator abstracts user validation efforts
+type Validator struct {
+	Login    string
+	UserName string
+	Password string
+}
+
+var (
+	usernameExp = regexp.MustCompile("^([a-zA-Z]+[0-9a-zA-Z-_ ,'.]*|)$")
+	loginExp    = regexp.MustCompile("^[a-zA-Z]+[0-9a-zA-Z-_.]*$")
+)
+
+// NewValidator creates/allocates a new user validation
+func NewValidator(login string,
```

Comment:
Since we only return nil for the error, and all of the calls to NewValidator always ignore the error, please just drop returning an error and only return the validator.

Reply:
Makes sense..Since we dont perform any checks at object creation..this can b removed

Reply:
Done